internal/gengapic/paging.go

```diff
@@ -161,6 +170,69 @@ func (g *generator) pagingField(m *descriptor.MethodDescriptorProto) (*descripto
 	return elemFields[0], nil
 }
 
+// diregapicPagingField reports the "resource field" to be iterated over by paginating method m
+// and the "num elements" field that tells the server the maximum number of elements to return per page.
+// Makes particular allowance for diregapic idioms: maps can be paginated over,
+// and either 'page_size' XOR 'max_results' are allowable fields in the request.
+func (g *generator) diregapicPagingField(m *descriptor.MethodDescriptorProto) (*descriptor.FieldDescriptorProto, *descriptor.FieldDescriptorProto, error) {
```

Comment:
I would consider changing the name to something like `getPagingFields` because:
- plural: you're getting more than one field
- diregapic: this returns fields for either the diregapic or non-diregapic case; ie it handles both scenarios

Reply:
(or, in the spirit of the previous comment, making this essentially be _the_ `pagingFields()` function.

Reply:
Good point; named returns sound like the correct thing here.
You're right: the diregapic paging fields are a superset of the 'regular' paging fields, so it might make sense to use the same function for both. The one exception I can think of is if the response has multiple repeated fields. I'll think about that some more.